src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ExecutionServiceCluster.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.dsde.rawls.dataaccess
+
+import org.broadinstitute.dsde.rawls.RawlsException
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkflowRecord
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.concurrent.{Future}
+import scala.util.Try
+
+/**
+  * Created by davidan on 6/14/16.
+  */
+trait ExecutionServiceCluster extends ErrorReportable {
+  override val errorReportSource = "cromwell"
+
+  // ====================
+  // facade methods
+  // ====================
+  def submitWorkflow(wdl: String, inputs: String, options: Option[String], userInfo: UserInfo): Future[ExecutionServiceStatus]
+
+  def submitWorkflows(workflowRecs: Seq[WorkflowRecord], wdl: String, inputs: Seq[String], options: Option[String], userInfo: UserInfo): Future[(ExecutionServiceId, Seq[Either[ExecutionServiceStatus, ExecutionServiceFailure]])]
+
+  def status(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceStatus]
+
+  def callLevelMetadata(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionMetadata]
+
+  def outputs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceOutputs]
+
+  def logs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceLogs]
+
+  def abort(wfe: WorkflowExecution, userInfo: UserInfo): Future[Try[ExecutionServiceStatus]]
+
+}
+
+case class WorkflowExecution(
+  id: String,
+  executionServiceId: Option[String]
+)
+
+// we need the external id in order to query cromwell, and the execution service id in order to route to the right instance
+object WorkflowExecution {
```

Comment:
re: kosher ... well, it works! WorkflowExecution isn't used in case statements; I'm just in the habit of using case classes to get all the boilerplate (e.g. equals, toString). I can revert to a regular class.